pkg/helm/client.go

```diff
@@ -258,296 +208,439 @@ func (h *HelmClient) PullByRefWithRetries(ctx context.Context, ref string, versi
+	// Build helm install command arguments
+	args := []string{"install", opts.ReleaseName}
+
+	// Handle chart source
+	chartPath, err := h.resolveChartPath(ctx, opts.ReleaseName, opts.ChartPath, opts.ChartVersion)
 	if err != nil {
-		return nil, fmt.Errorf("load chart: %w", err)
+		return "", fmt.Errorf("resolve chart path: %w", err)
+	}
+	args = append(args, chartPath)
+
+	// Add namespace
+	if opts.Namespace != "" {
+		args = append(args, "--namespace", opts.Namespace)
+		args = append(args, "--create-namespace")
 	}
 
-	if req := chartRequested.Metadata.Dependencies; req != nil {
-		if err := action.CheckDependencies(chartRequested, req); err != nil {
-			return nil, fmt.Errorf("check chart dependencies: %w", err)
+	// Add wait options
+	args = append(args, "--wait")
+	args = append(args, "--wait-for-jobs")
+
+	// Add timeout
+	timeout := opts.Timeout
+	if timeout == 0 {
+		timeout = 5 * time.Minute
+	}
+	args = append(args, "--timeout", timeout.String())
+
+	// Add replace flag
+	args = append(args, "--replace")
+
+	// Add values if provided
+	if opts.Values != nil {
+		valuesFile, err := h.createValuesFile(opts.Values)
+		if err != nil {
+			return "", fmt.Errorf("create values file: %w", err)
 		}
+		defer os.Remove(valuesFile)
+		args = append(args, "--values", valuesFile)
 	}
 
-	cleanVals, err := cleanUpGenericMap(opts.Values)
-	if err != nil {
-		return nil, fmt.Errorf("clean up generic map: %w", err)
+	// Add labels if provided
+	if opts.Labels != nil {
+		var labelPairs []string
+		for k, v := range opts.Labels {
+			labelPairs = append(labelPairs, fmt.Sprintf("%s=%s", k, v))
+		}
+		args = append(args, "--labels", strings.Join(labelPairs, ","))
 	}
 
-	release, err := client.RunWithContext(ctx, chartRequested, cleanVals)
+	// Add kubeconfig and context if available
+	args = h.addKubernetesEnvArgs(args)
+
+	// Add debug flag to report progress and capture debug logs
+	args = append(args, "--debug")
+
+	// NOTE: we don't set client.Atomic = true on install as it makes installation failures difficult to debug
+	// since it will rollback the release.
+
+	// Execute helm install command
+	stdout, stderr, err := h.executor.ExecuteCommand(ctx, nil, args...)
 	if err != nil {
-		return nil, fmt.Errorf("helm install: %w", err)
+		return "", fmt.Errorf("helm install: %w, stderr: %s", err, stderr)
 	}
 
-	return release, nil
+	return stdout, nil
 }
 
-func (h *HelmClient) Upgrade(ctx context.Context, opts UpgradeOptions) (*release.Release, error) {
-	cfg, err := h.getActionCfg(opts.Namespace)
+// resolveChartPath handles chart source resolution for install, upgrade, and render operations
+func (h *HelmClient) resolveChartPath(ctx context.Context, releaseName, chartPath, chartVersion string) (string, error) {
+	if h.airgapPath != "" {
+		// Use chart from airgap path
+		return filepath.Join(h.airgapPath, fmt.Sprintf("%s-%s.tgz", releaseName, chartVersion)), nil
+	}
```

Comment:
this never felt like it belonged in this package. perhaps something to address later